release/train_tests/benchmark/config.py

```diff
@@ -35,15 +38,21 @@ class BenchmarkConfig(BaseModel):
     dataloader_config: DataLoaderConfig = Field(
         default_factory=lambda: DataLoaderConfig(),
     )
+    num_torch_workers: int = 0
+    torch_dataloader_timeout_seconds: int = 300
+    torch_pin_memory: bool = True
+    torch_non_blocking: bool = True
```

Comment:
I have this nested `DataLoaderConfig` class that swaps depending on the `dataloader_type` that is set. For example, we could add a `DataloaderType.TORCH_S3_DATALOADER` option for the one you're adding, then put all the torch dataloader specific arguments into a `TorchDataloaderConfig`.

That way, only the relevant configs are available when a certain dataloder type is set (ex: it doesn't make sense to access torch dataloader configs when using ray data)

https://github.com/ray-project/ray/blob/a3d571920ed181d25c281fbaf5e83615f9383347/release/train_tests/benchmark/config.py#L7-L19